socks/socks4.py

```diff
@@ -19,11 +19,20 @@ class SOCKS4Command(bytes, enum.Enum):
 class SOCKS4Request(typing.NamedTuple):
     command: SOCKS4Command
     port: int
-    addr: str
+    addr: bytes
     user_id: bytes
 
     def dumps(self) -> bytes:
-        raise NotImplementedError()
+        return b"".join(
+            [
+                bytes([4]),
+                bytes(self.command),
```

Comment:
Do we have to have a `bytes()` call here or can we add `__bytes__()` onto `SOCKS4Command` and all the other `(bytes, Enum)` classes?